plugins/inputs/smart/smart.go

```diff
@@ -373,6 +391,37 @@ func gatherDisk(acc telegraf.Accumulator, timeout internal.Duration, usesudo, co
 				}
 			}
 		}
+
+		if m.ErrorCounterLog {
+			ecl := errorCounterLog.FindStringSubmatch(line)
+			if len(ecl) == 9 {
+				eclTags := map[string]string{}
+				eclFields := make(map[string]interface{})
+
+				eclTags["page_name"] = ecl[1]
+
+				keys := [...]string{"device", "model", "serial_no", "wwn", "capacity", "enabled"}
+				for _, key := range keys {
+					if value, ok := deviceTags[key]; ok {
+						eclTags[key] = value
+					}
+				}
+
+				eclFields["errors_corrected_by_eccfast"] = ecl[2]
+				eclFields["errors_corrected_by_eccdelayed"] = ecl[3]
+				eclFields["errors_corrected_by_rereads_rewrites"] = ecl[4]
+				eclFields["total_errors_corrected"] = ecl[5]
+				eclFields["correction_algorithm_invocations"] = ecl[6]
+				gigabytes_processed, err := strconv.ParseFloat(ecl[7], 64)
+				if err != nil {
+					gigabytes_processed = 0
+				}
+				eclFields["gigabytes_processed"] = gigabytes_processed * 1000 * 1000 * 1000
```

Comment:
Isn't this now the `bytes_processed`? And are we talking _GiB_ (base 2) or _GB_ (base 10)?

Reply:
You are right this values is no longer expressed as gigabytes therefore maybe it's better to keep it as gigabytes. However it's provided by smartctl output with precision down to three decimal places. Need to think how to handle this:

* send it as float
* turn it into bytes or megabytes
* round it to neareset gigabyte (loosing precision)

Looking at the output of smartctl command it states that this field is `10^9 bytes` therefore I assume it's GB (base 10).

Reply:
I would either go for the bytes (option 2) or the float option. Loosing precision is kind of the last resort...